packages/legend-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -38,15 +42,165 @@ import {
-import { EntityDiff, Revision } from '@finos/legend-server-sdlc';
+import { Entity } from '@finos/legend-model-storage';
+import {
+  EntityDiff,
+  EntityChange,
+  Revision,
+  EntityChangeType,
+} from '@finos/legend-server-sdlc';
 import { LEGEND_STUDIO_LOG_EVENT_TYPE } from '../LegendStudioLogEvent';
 
+class PatchChangeState {
+  editorStore: EditorStore;
+  sdlcState: EditorSDLCState;
+
+  changes: EntityChange[] | undefined;
+  currentChanges: EntityChange[] = [];
+  isLoadingChanges = false;
+  modal = false;
+  isValidPatch = false;
+
+  constructor(editorStore: EditorStore, sdlcState: EditorSDLCState) {
+    makeAutoObservable(this, {
+      editorStore: false,
+      sdlcState: false,
+      setModal: action,
+      openModal: action,
+      closeModal: action,
+      deleteChange: action,
+      loadPatchFile: flow,
+    });
+
+    this.editorStore = editorStore;
+    this.sdlcState = sdlcState;
+  }
+
+  openModal(localChanges: EntityChange[]): void {
+    this.currentChanges = localChanges;
+    this.setModal(true);
+  }
+
+  closeModal(): void {
+    this.currentChanges = [];
+    this.setPatchChanges(undefined);
+    this.setModal(false);
+  }
+
+  setModal(val: boolean): void {
+    this.modal = val;
+  }
+
+  setIsValidPatch(val: boolean): void {
+    this.isValidPatch = val;
+  }
+
+  setPatchChanges(changes: EntityChange[] | undefined): void {
+    this.changes = changes;
+  }
+
+  deleteChange(change: EntityChange): void {
+    if (this.changes) {
+      deleteEntry(this.changes, change);
+    }
+  }
+
+  get overiddingChanges(): EntityChange[] {
+    if (this.changes?.length) {
+      return this.changes.filter((change) =>
+        this.currentChanges.find(
+          (local) => local.entityPath === change.entityPath,
+        ),
+      );
+    }
+    return [];
+  }
+
+  *loadPatchFile(file: File): GeneratorFn<void> {
+    try {
+      this.setPatchChanges(undefined);
+      assertTrue(
+        file.type === ContentType.APPLICATION_JSON,
+        'Patch file expected to be of type `JSON`',
+      );
+      const fileText = (yield readFileAsText(file)) as string;
+      const entityChanges = JSON.parse(fileText) as {
+        entityChanges: PlainObject<EntityChange>[];
+      };
+      const changes = entityChanges.entityChanges.map((e) =>
+        EntityChange.serialization.fromJson(e),
+      );
+      this.setPatchChanges(changes);
+      this.setIsValidPatch(true);
+    } catch (error) {
+      assertErrorThrown(error);
+      this.setIsValidPatch(false);
+      this.editorStore.applicationStore.notifyError(
+        `Error loading patch: ${error.message}`,
+      );
+    }
+  }
+
+  *applyChanges(): GeneratorFn<void> {
+    if (this.changes?.length) {
+      this.editorStore.graphState.loadEntityChangesToGraph(this.changes);
+      this.closeModal();
+    }
+  }
+
+  applyEntityChanges(
+    currentEntities: Entity[],
+    entityChanges: EntityChange[],
+  ): Entity[] {
+    entityChanges.forEach((change) => {
```

Comment:
I start to think if `classifierPath` and `content` not provided, we probably should filter out those element, the patch is not proper then. What do you think?

Reply:
for content it makes sense. Should we do this for classifierPath? not sure since it would load fine without it.